ginivision/src/main/java/net/gini/android/vision/review/ReviewFragmentImpl.java

```diff
@@ -96,81 +115,183 @@ public void onDocumentAnalyzed() {
 
     public void onCreate(@Nullable Bundle savedInstanceState) {
         forcePortraitOrientationOnPhones(mFragment.getActivity());
+        if (mDocument.getType() == Document.Type.PDF) {
+            mListener.onProceedToAnalysisScreen(mDocument);
+            return;
+        }
         if (savedInstanceState != null) {
             restoreSavedState(savedInstanceState);
-            LOG.info("Should analyze document");
-            mListener.onShouldAnalyzeDocument(Document.fromPhoto(mPhoto));
-        } else {
-            applyCompressionToJpeg(new PhotoEdit.PhotoEditCallback() {
-                @Override
-                public void onDone(@NonNull Photo photo) {
-                    if (mNextClicked || mDocumentWasModified || mStopped) {
-                        return;
-                    }
-                    LOG.info("Should analyze document");
-                    mListener.onShouldAnalyzeDocument(Document.fromPhoto(mPhoto));
-                }
-
-                @Override
-                public void onFailed() {
-                    if (mNextClicked || mStopped) {
-                        return;
-                    }
-                    LOG.error("Failed to compress the jpeg");
-                    mListener.onError(new GiniVisionError(GiniVisionError.ErrorCode.REVIEW,
-                            "An error occurred while compressing the jpeg."));
-                }
-            });
         }
     }
 
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-        View view = inflater.inflate(R.layout.gv_fragment_review, container, false);
+            Bundle savedInstanceState) {
+        final View view = inflater.inflate(R.layout.gv_fragment_review, container, false);
         bindViews(view);
         setInputHandlers();
-        observeViewTree(view);
         return view;
     }
 
     public void onStart() {
-        showDocument();
         mNextClicked = false;
         mStopped = false;
+        if (mPhoto == null) {
+            final Activity activity = mFragment.getActivity();
+            if (activity == null) {
+                return;
+            }
+            showActivityIndicator();
```

Comment:
Review Screen contains an activity indicator in the middle and is shown until the preview created and shown.